### DISCUSSION TITLE
Nylo Improvements v5 [Open Discussion]

### DISCUSSION THREAD

**Discussion by agordn52** [nylo-core/nylo#78] [Ideas]
## Nylo Improvements

Here's a list of improvements that I think Nylo could benefit from. 

If you have any ideas, feel free to drop a comment 👍

---

- **Pages**
    It would be cool if we could have one class to handle building the widget rather than a StatefulWidget and State.
    Here's an example below.
``` dart 
class HomePage extends NyPage {

    static String path = "/home-page";
  
    @override
    Widget build(BuildContext context) {
    return Scaffold();
  }
}
```
- **Metro**
  - Use `metro make:theme` to automatically add a theme in the source code (currently you can create a theme but you have to manually add it into the config/theme.dart file)
  - New `metro make:interceptor` method to create interceptors in the **app/networking/dio/interceptors** directory
- **Renaming projects**
    - Easy way to 'Rename' project for IOS and Android. I found this package https://pub.dev/packages/rename which is useful, I think making this default would help developers
- **Nylo** [website](https://github.com/nylo-core/nylo.dev)
    - Search bar on the [docs](https://nylo.dev/docs/5.x) page to make it easier to find sections
- **New packages/services** 
   - Handle IAP (in app purchases)
   - Handle push notifications
- **Localization**
    -   Ability to auto-detect which locale to use

**Comment by bhargavraviya** [comment]
@agordn52 
Thank you for helping with your package.

Yes, it's an awesome package of renaming to help rename Android and iOS app names and package names.


What is new in package service?

- Handle IAP (in-app purchases).
- Handle push notifications (local notification or Firebase message package?)
- also add permission hander (https://pub.dev/packages/permission_handler)

please give more detail on the package/services.

**Comment by agordn52** [comment]
I had an idea for creating Models from json objects using Metro.

We could have a new flag on the **make:model** command e.g. `metro make:model car -json`

Then, it would prompt you to paste a JSON object into the CLI, for example.

``` dart 
{
      "brand": "Mercedes-Benz",
      "model": "Citan",
      "car body": "van",
      "color": "red",
      "price": "10000$",
      "year": 2010
}
```

And then using the `json_dart_generator` package in the [framework](https://github.com/nylo-core/framework), it could generate the class and create the file into the project.
